lib/queue.js

```diff
@@ -423,7 +430,10 @@ Queue.prototype.run = function(concurrency){
   var promises = [];
   var _this = this;
 
-  return this.processStalledJobs().then(function(){
+  // In case of connection loss, running `processStalledJobs` will repick jobs here.
+  var start = this.opts.processStalledJobs ? this.processStalledJobs() : Promise.resolve();
```

Comment:
is this still needed to be run when the queue starts, instead of just letting it start after `_this.LOCK_RENEW_TIME` ?